src/pip/_internal/operations/freeze.py

```diff
@@ -200,6 +201,15 @@ def get_requirement_info(dist):
             )
         ]
         return (location, True, comments)
+    except RemoteNotValidError as ex:
+        url, = ex.args
+        req = dist.as_requirement()
+        comments = [
+            f"# Editable {type(vcs_backend).__name__} install ({req}) with "
+            f"either a deleted local remote or invalid URI:",
+            f"# '{url}'",
```

Comment:
Hmm, I'm confusing the hell out of myself with all these local repos with remotes to other local repos which may or may not exist. In the test example in `test_freeze.py::test_freeze_remote` there are two different urls but I'm not sure if that's really the correct output.
```
# Editable Git install (version-pkg==0.1) with either a deleted local remote or invalid URI:
# '/tmp/pytest-of-brenainn/pytest-0/test_freeze_git_remote0/workspace/scratch/version_pkg-other'
-e /tmp/pytest-of-brenainn/pytest-0/test_freeze_git_remote0/workspace/scratch/pip-test-package
```
Hang on whilst I untangle my head...

Reply:
Ok, so the 2<sup>nd</sup> one is the path to the editably-installed package. This matches what `pip freeze` does when there is no remote so all is good. The 1<sup>st</sup> commented out line is the invalid remote. i.e. The output of:
```bash
git -C 'url number 2' remote -v
```
which isn't strictly necessary but I thought it would help to explain to the user what pip is unhappy about.